test/yaml_edit/test_case.dart

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:pub/src/yaml_edit/yaml_edit.dart';
+import 'package:test/test.dart';
+import 'package:yaml/yaml.dart';
+
+import 'test_utils.dart';
+
+/// Interface for creating golden Test cases
+class TestCases {
+  final List<TestCase> testCases;
+
+  /// Creates a [TestCases] object based on test directory and golden directory
+  /// path.
+  static Future<TestCases> getTestCases(Uri testDirUri, Uri goldDirUri) async {
+    var testDir = Directory.fromUri(testDirUri);
+    var testCaseList = [];
+
+    if (testDir.existsSync()) {
+      /// Recursively grab all the files in the testing directory.
+      var entityStream = testDir.list(recursive: true, followLinks: false);
+      entityStream =
+          entityStream.where((entity) => entity.path.endsWith('.test'));
+
+      var testCasesPathStream = entityStream.map((entity) => entity.uri);
+      var testCasePaths = await testCasesPathStream.toList();
+
+      testCaseList = testCasePaths.map((inputUri) {
+        var inputName = inputUri.toFilePath(windows: false).split('/').last;
+        var inputNameWithoutExt = inputName.substring(0, inputName.length - 5);
```

Comment:
Similarly, consider what `package:path` can already do... `pub` already has a dependency upon it.